docs/promtail-failure-modes.md

```diff
@@ -0,0 +1,57 @@
+# Promtail Failure Modes
+
+This document describes known failure modes of `promtail` on edge cases and the adopted trade-offs.
+
+
+## A tailed file is truncated while `promtail` is not running
+
+Given the following order of events:
+
+1. `promtail` is tailing `/app.log`
+2. `promtail` current position for `/app.log` is `100` (bytes)
+3. `promtail` is stopped
+4. `/app.log` is truncated and new logs are appended to it
+5. `promtail` is restarted
+
+When `promtail` is restarted, it reads the previous position (`100`) from the positions file. Two scenarios are then possible:
+
+- `/app.log` size is < than the position before truncating
+- `/app.log` size is >= than the position before truncating
+
+If the `/app.log` file size is less than the previous position, then the file is detected as truncated and logs will be tailed starting from position `0`. Otherwise, if the `/app.log` file size is >= than the previous position, `promtail` can't detect it was truncated while not running and will continue tailing the file from position `100`.
+
+
+## Loki ingester is unavailable
```

Comment:
it's not specifically the ingester (would actually be any component in the pathway to loki, gateway/distributor, etc)

So may "Loki is unavailable"